sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala

```diff
@@ -94,6 +94,30 @@ abstract class SQLViewSuite extends QueryTest with SQLTestUtils {
     }
   }
 
+  test("create a permanent view on a temp view used inside an expression") {
+    withTable("base_table") {
+      withTempView("temp_view") {
+        sql("CREATE TABLE base_table (id int) USING parquet")
+        sql("CREATE TEMPORARY VIEW temp_view AS SELECT * FROM base_table")
+
+        val statements = Seq(
+          "CREATE VIEW v AS SELECT * FROM base_table WHERE EXISTS (SELECT * FROM temp_view)",
+          "CREATE VIEW v AS SELECT id, (SELECT count(*) FROM temp_view) FROM base_table",
+          "CREATE VIEW v AS SELECT * FROM base_table " +
+            "WHERE id IN (SELECT id FROM base_table WHERE EXISTS (SELECT 1 FROM temp_view))"
```

Comment:
can we move them to `create_view.sql`? That's a better plan for end-to-end tests.

Reply:
I moved these to `create_view.sql`. The first two scenarios were already there, so I just moved the last one. thanks for the feedback.